app/javascript/packs/MainApp/components/RefereeProfile.jsx

```diff
@@ -148,12 +155,60 @@ class RefereeProfile extends Component {
       .catch(this.setErrorStateFromBackendData);
   };
 
-  hasPassedTest(level) {
-    const { referee: { certifications } } = this.state;
+  hasPassedTest = (level) => {
+    const { referee: { certifications } } = this.state
 
     return certifications.some(({ level: certificationLevel }) => certificationLevel === level)
   }
 
+  handlePaymentSuccess = (payment) => {
+    const { paid } = payment
+
+    if (paid) {
+      this.setState({ paymentSuccess: true })
+      axios
+        .patch(this.currentRefereeApiRoute, {
+          submitted_payment_at: DateTime.local().toString()
```

Comment:
Is there a way to validate the payment on server side to not having to trust the client?

Reply:
Not unless we want to do a full api integration. And at this point we don't really have the time. At the end of the day all we need to know is that paypal got the payment successfully. If, later on, we had to give a refund or whatever we can always set the attribute to nil to force another payment. Trust me, this isn't my preferred way of doing things, but it securely gets the job done on the paypal side, without us having to roll our own billing structure.